include/sparrow/typed_array.hpp

```diff
@@ -0,0 +1,358 @@
+    class typed_array
+    {
+    public:
+
+        using layout_type = L;
+
+        using reference = typename layout_type::reference;
+        using const_reference = typename layout_type::const_reference;
+
+        using iterator = typename layout_type::iterator;
+        using const_iterator = typename layout_type::const_iterator;
+
+        using size_type = typename layout_type::size_type;
+        using const_bitmap_range = typename layout_type::const_bitmap_range;
+        using const_value_range = typename layout_type::const_value_range;
+
+        explicit typed_array(array_data data);
+
+        // Element access
+
+        ///@{
+        /*
+         * @brief Access specified element with bounds checking.
+         *
+         * @description Returns a reference to the element at the specified index \p i, with bounds checking.
+         * If \p i is not within the range of the container, an exception of type std::out_of_range is thrown.
+         *
+         * @param i The index of the element to access.
+         * @return A reference to the element at the specified index.
+         * @throws std::out_of_range if i is out of range.
+         */
+        reference at(size_type i);
+        const_reference at(size_type i) const;
+        ///@}
+
+        ///@{
+        /*
+         * @brief Access specified element.
+         *
+         * @description Returns a reference to the element at the specified index \p i. No bounds checking is
+         * performed.
+         *
+         * @param i The index of the element to access.
+         * @return A reference to the element at the specified index.
+         */
+        reference operator[](size_type);
+        const_reference operator[](size_type) const;
+        ///@}
+
+        ///@{
+        /*
+         * @brief Access the first element.
+         *
+         * @description Returns a reference to the first element.
+         * Calling front on an empty container is undefined.
+         *
+         * @return A reference to the first element.
+         */
+        reference front();
+        const_reference front() const;
+        ///@}
+
+        ///@{
+        /*
+         * @brief Access the last element.
+         *
+         * @description Returns a reference to the last element.
+         * Calling back on an empty container is undefined.
+         *
+         * @return A reference to the last element.
+         */
+        reference back();
+        const_reference back() const;
+        ///@}
+
+        // Iterators
+
+        ///@{
+        /*
+         * @brief Returns an iterator to the beginning.
+         *
+         * @description Returns an iterator to the first element of the vector.
+         * If the vector is empty, the returned iterator will be equal to end().
+         *
+         * @return An iterator to the beginning.
+         */
+        iterator begin();
+        const_iterator cbegin() const;
```

Comment:
The const overload of `begin` is missing (you generally need it in generic code that accepts both const and non const objects).